shared/roughtime/roughtime.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+// Package roughtime is a wrapper for a roughtime clock source
+package roughtime
+
+import (
+	"fmt"
+	"time"
+	"encoding/base64"
+
+	"github.com/cloudflare/roughtime"
+	"roughtime.googlesource.com/roughtime.git/go/config"
+)
+
+// offset is the difference between the system time and the time returned by
+// the roughtime server
+var offset time.Duration
+
+func init() {
+	pk, err := base64.StdEncoding.DecodeString("gD63hSj3ScS+wuOeGrubXlq35N1c5Lby/S+T7MNTjxo=")
```

Comment:
What is this private key? Should it be provided by the user?

Reply:
This is the public key (not private) used to check responses from the roughtime server roughtime.cloudflare.com

github.com/cloudflare/roughtime can load this from a config file. If this PR works as expected, we can add a basic roughtime config file in Prysm and load it on startup.

Reply:
Thanks for clarifying. It would be helpful to put a comment above this line to explain what this public key is and where it originated from. 

I'm not sure if we need it to be dynamically loaded, what do you think?

Reply:
It may be better to load this config file containing addresses and keys of multiple servers to not have a single point of failure.